### PR TITLE
[coop] Separate host and target suspend policies

### PR DESCRIPTION
**Pull request by lambdageek**
Always set target suspend policy to coop when cross-compiling to watchOS.

Target policy affects two places:
- safepoint generation in AOT in mini
- blocking transition insertion in marshal-ilgen

Everything else (primarily metadata) cares about host policy (same as before).

There's one tricky bit:  if `MONO_ENABLE_BLOCKING_TRANSITION` env var is set, it affects both host and target, for backward compatibility.  (although this variable wasn't widely publicized or used in production, afaik).

This addresses #11765 which was caused by mono/monodevelop#6327 where `MONO_THREADS_SUSPEND=preemptive` affected both host policy and the watchOS cross compiler target.  Now the env var is just about the host.

### CODE REVIEW

**Comment by vargaz** [comment]
We should avoid safepoints at least on wasm which has no async gc.

**Comment by luhenry** [comment]
@vargaz this is a much smaller problem, and it would be straightforward to not generate safepoints on wasm (the `if (mono_threads_polling) mono_threads_state_poll()` code)., but still allow us to unify the paths in, for example, `marshal-ilgen.c`.

**Comment by lambdageek** [comment]
> but still allow us to unify the paths in, for example, marshal-ilgen.c

I don't understand this part: wasm doesn't need thread state transitions either.

File: mono/utils/mono-threads-coop.c [hunk 1]
```diff
@@ -640,40 +639,66 @@ mono_threads_suspend_policy_name (void)
...
+	static int inenv = -1;
+	if (G_UNLIKELY (inenv == -1))
+		inenv = g_hasenv ("MONO_ENABLE_BLOCKING_TRANSITION");
+	return inenv;
+}
+
+static gboolean
+blocking_transition_from_policy (MonoThreadsSuspendPolicy p)
+{
+	switch (mono_threads_host_suspend_policy ()) {
```
Comment (lambdageek) [inline mono/utils/mono-threads-coop.c]:
```suggestion
	switch (p) {
```

**Comment by lambdageek** [comment]
This is going to be reworked - blocking transitions and safepoints on most targets (except wasm).

**Pull request closed without merging**